src/util/deps.ts

```diff
@@ -0,0 +1,52 @@
+import { INestApplicationContext } from "@nestjs/common"
+import { ModuleRef } from "@nestjs/core"
+import { useLogger } from "./logger"
+
+interface Type<T> extends Function {
+  new (...args: any[]): T;
+}
+type MaybePromise<T> = T | Promise<T> 
+
+let app: INestApplicationContext | undefined
+const buffering: Array<(ctx: INestApplicationContext) =>MaybePromise<unknown>> = []
+
+export function setGlobalApplicationContext(context: INestApplicationContext) {
+  if (app) {
+    throw new Error('ApplicationContext can not be set twice.')
+  }
+  app = context
+
+  while (buffering.length) {
+    const fn = buffering.shift()
+    if (fn) fn(context)
+  }
+}
+
+export function runWithDeps<RESULT>(
+  deps: Array<Type<unknown> | string | symbol>,
+  func: (...args: Array<string | symbol | any>) => MaybePromise<RESULT>,
+) {
+  const logger = useLogger('System')
+  const exec = (ctx: INestApplicationContext) => {
+    const moduleRef = ctx.get(ModuleRef)
+    return func(...deps.map((dep) => moduleRef.get(dep, { strict: false })))
+  }
+
+  if (app) return exec(app)
```

Comment:
This is interesting, could you explain what's the case that codes run after this line?

I saw the `app` is set from main.ts.